$NetBSD: patch-configure.ac,v 1.1 2023/07/14 10:43:34 micha Exp $

Upstream expects a specific compiler/linker for a specific OS.
pkgsrc is not limited to such configurations.
Use COMPILER_RPATH_FLAG to set runtime library search path.

--- configure.ac.orig	2023-07-06 02:10:20.000000000 +0000
+++ configure.ac
@@ -5040,18 +5040,18 @@ case "${canonical}" in
         dirname=`which gcc`
         dirname=`dirname $dirname`
         dirname=`dirname $dirname`/lib
-	if test "$dirname" != "/lib"; then
-  	    XLOCK_LDFLAGS="${XLOCK_LDFLAGS}:$dirname"
-  	    XMLOCK_LDFLAGS="${XMLOCK_LDFLAGS}:$dirname"
-	fi
+#	if test "$dirname" != "/lib"; then
+#  	    XLOCK_LDFLAGS="${XLOCK_LDFLAGS}:$dirname"
+#  	    XMLOCK_LDFLAGS="${XMLOCK_LDFLAGS}:$dirname"
+#	fi
     fi
     case "${canonical}" in
       *-*-freebsd* )
         FORTUNE_PATH="/usr/bin/fortune"
       ;;
     esac
-    XLOCKLDFLAGS="-Wl,-R${XLOCK_LDFLAGS}"
-    XMLOCKLDFLAGS="-Wl,-R${XMLOCK_LDFLAGS}"
+    XLOCKLDFLAGS="${COMPILER_RPATH_FLAG}${XLOCK_LDFLAGS}"
+    XMLOCKLDFLAGS="${COMPILER_RPATH_FLAG}${XMLOCK_LDFLAGS}"
   ;;
 
   *-*-dgux5* )
@@ -5169,13 +5169,13 @@ dnl    XLOCKLIBS="${XLOCKLIBS} -lsocket 
         dirname=`which gcc`
         dirname=`dirname $dirname`
         dirname=`dirname $dirname`/lib
-	if test "$dirname" != "/lib"; then
-  	    XLOCK_LDFLAGS="${XLOCK_LDFLAGS}:$dirname"
-  	    XMLOCK_LDFLAGS="${XMLOCK_LDFLAGS}:$dirname"
-	fi
+#	if test "$dirname" != "/lib"; then
+#  	    XLOCK_LDFLAGS="${XLOCK_LDFLAGS}:$dirname"
+#  	    XMLOCK_LDFLAGS="${XMLOCK_LDFLAGS}:$dirname"
+#	fi
     fi
-    XLOCKLDFLAGS="-R${XLOCK_LDFLAGS}"
-    XMLOCKLDFLAGS="-R${XMLOCK_LDFLAGS}"
+    XLOCKLDFLAGS="${COMPILER_RPATH_FLAG}${XLOCK_LDFLAGS}"
+    XMLOCKLDFLAGS="${COMPILER_RPATH_FLAG}${XMLOCK_LDFLAGS}"
   ;;
 
   *-*-sunos4* )
